contributing/core/developing-using-local-app.md

```diff
@@ -0,0 +1,76 @@
+# Developing Using Your Local Version of Next.js
+
+There are two options to develop with your local version of the codebase:
+
+## Develop inside the monorepo
+
+This will use the version of `next` built inside of the Next.js monorepo. You can also let `pnpm dev` run in a separate terminal. This will let you make changes to Next.js at the same time (note that some changes might require re-running `pnpm next-with-deps` or `pnpm next`).
+
+If your app does not have dependencies, you can create a directory inside the monorepo (eg.: `dev-app`) and run `pnpm next ./dev-app` without creating a `package.json` file.
+
+If you already have an app and it has dependencies, you can follow these steps:
+
+1. Move your app inside of the Next.js monorepo.
+
+2. Run with `pnpm next-with-deps ./app-path-in-monorepo`
+
+## Set as a local dependency in package.json
```

Comment:
@SukkaW let's continue https://github.com/vercel/next.js/pull/39778#discussion_r950725928 here.

In my opinion, this whole section is cumbersome and seems to give a bad DX (see also #40514).

Maybe we should just remove it and recommend developing inside the monorepo exclusively?

When I am trying to reproduce issues, it is very often not a package manager-related issue, so I just pull their repository inside the monorepo and use either `pnpm next` or `pnpm next-with-deps`.

Reply:
@balazsorban44 Sometimes I do encounter the issue (#40514) as well, and I agree it is very annoying.

However, *personally* I don't like the idea of messing with the local `next.js` repo folder either. It could mess with git, `package.json`, or pnpm workspace.

Reply:
It can be shorter.
```bash
pnpm rm next react react-dom
pnpm add ./path/to/next.js/{packages/next,node_modules/{react,react-dom}}
```
This does all of steps 1-3, and step 5 is no longer required anyway (see https://github.com/vercel/next.js/pull/39778#discussion_r950726281) so the whole section is basically just these two lines.

And if this section is shortened like this, it would probably be fine to recombine the file with developing.md.

Reply:
Also, in most cases, the issue is not related to `react` or `react-dom`, or at least not related to behavior changes between React 17 & 18. Linking `react` and `react-dom` is good enough for *most* cases.

Reply:
Indeed, this seems to be much more simple, I just tested it. :+1: Thanks for the feedback to both of you! I updated the PR.